dialogue-core/src/main/metrics/dialogue-core-metrics.yml

```diff
@@ -8,12 +8,21 @@ namespaces:
     metrics:
       response:
         type: timer
-        tags: [channel-name, service-name]
-        docs: Request time, note that this does not include time spent reading the response body.
-      response.error:
-        type: meter
-        tags: [channel-name, service-name, reason]
-        docs: Rate of errors received by reason and service-name. Currently only errors with reason `IOException` are reported.
+        tags: [channel-name, service-name, endpoint, status]
+        docs: |
+          Request time split by status. These metrics and their definitions are mostly useful to dialogue developers,
+          to measure the success of routing and retry strategies. Possible status values are:
+          * success: always excludes time spent reading the response body. This classifies requests,
+            which we successfully routed, maybe retried and returned to the user.
+            This does not mean that the ultimate user might not see an exception, e.g. 404, 401 will still potentially
+            result in a user visible failure.
+          * preventable_failure:
+            - QoS failures (429, 308, 503)
+            - 500 requests for endpoints that are retryable.
+            - IO related failures that we retry: e.g. UnknownHostException, connect timeouts, SSLException.
+          * other_failure: things we can't do anything about, e.g.
+            - 500 requests which we cannot retry
+            - read/write timeouts.
```

Comment:
I'm not sure about categorizing `preventable_failures` differently from `other_failures`, we may get diminishing returns for the metric-name fanout at that point.

Reply:
Done